crates/rpc/src/methods/send_transaction.rs

```diff
@@ -189,12 +165,26 @@ impl SendTransaction {
             .parse()
             .map_err(|_| Error::CannotParseUrl(self.network.http_url.clone()))?;
 
-        let provider = ProviderBuilder::new()
-            .with_recommended_fillers()
-            .wallet(signer.to_wallet())
-            .on_http(url);
+        if self.network.is_dev() {
```

Comment:
for now (as in, before we do the high nonce thing) this is probably not enough. I would go on the safe side and only allow this on 31337 chain ids